### DISCUSSION TITLE
Option to specify project-wide `jupyter:` kernel

### DISCUSSION THREAD

**Discussion by pat-alt** [quarto-dev/quarto-cli#2850] [Feature Requests]
Currently Jupyter kernels need to be specified on a per-document basis. In other words, specifying the Jupyter kernel in `_quarto.yml` or `_metadata.yml` has no effect: see this [example](https://github.com/pat-alt/quarto-example/tree/main/project-wide-jupyter-kernel) where the kernel defaults to the default kernel `julia-1.6` for the [document](https://github.com/pat-alt/quarto-example/blob/main/project-wide-jupyter-kernel/doc-without-specified-kernel.md) without `jupyter` explicitly declared in the YAML header. Is this a deliberate design choice? Is there anything that speaks against specifying kernels for multiple documents?

**Background**: I'm using Quarto to document my Julia packages. Being able to to control the Julia version for all my docs at once would give me a way to check locally if all code chunks execute successfully for that version.

**Comment by cscheid** [comment]
It should be possible to do something like this, and your document structure looks good to me. Feels like it's a bug on our side. I'll create an issue to track.

**Reply by cscheid** [reply]
@pat-alt thanks for the report. This is now fixed on `main`.

**Reply by pat-alt** [reply]
Thanks very much for the quick fix

**Comment by mcanouil** [comment]
Closing as completed and in favour of https://github.com/quarto-dev/quarto-cli/issues/2853.
